app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -8,14 +12,56 @@ class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d+)$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d+) (\d+) ?/ ?(\d+)$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
 
-  fun getFractionFromString(text: String): Fraction {
+  /**
+   * Returns a [FractionParsingError] for the specified text input if it's an invalid fraction, or
+   * [FractionParsingError.VALID] if no issues are found. Note that a valid fraction returned by this method is guaranteed
+   * to be parsed correctly by [parseRegularFraction].
+   *
+   * This method should only be used when a user tries submitting an answer. Real-time error detection should be done
+   * using [setPendingAnswerErrorOnRealTime], instead.
+   */
```

Comment:
Also add `@param` and `@return` to this kDoc which indicates what this function accepts and returns.

Reply:
This was suggested by ben please see the previous comments.

Reply:
Okay. then lets keep it this way only.